controllers/storagecluster/storagecluster_controller.go

```diff
@@ -233,12 +233,10 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&extv1.CustomResourceDefinition{},
 			enqueueStorageClusterRequest,
 			builder.WithPredicates(
-				util.NamePredicate(VirtualMachineCrdName),
-				util.CrdCreateAndDeletePredicate(&r.Log, VirtualMachineCrdName, r.AvailableCrds[VirtualMachineCrdName]),
-			),
-			builder.WithPredicates(
-				util.NamePredicate(StorageClientCrdName),
-				util.CrdCreateAndDeletePredicate(&r.Log, StorageClientCrdName, r.AvailableCrds[StorageClientCrdName]),
+				util.ComposePredicates(
+					util.CrdCreateAndDeletePredicate(&r.Log, VirtualMachineCrdName, r.AvailableCrds[VirtualMachineCrdName]),
+					util.CrdCreateAndDeletePredicate(&r.Log, StorageClientCrdName, r.AvailableCrds[StorageClientCrdName]),
+				),
```

Comment:
We could allow the NamePredicate to accept multiple CRD names and instead of using the ComposePredicates, we could use predicates.Or, which does a logical Or of the predicates. 
Also we should move to more generic EventTypePredicate as mentioned in the previous comment
```suggestion
				util.NamePredicate(VirtualMachineCrdName,StorageClientCrdName),
				predicate.Or(
					util.EventTypePredicate(
						!r.AvailableCrds[VirtualMachineCrdName],
						false,
						r.AvailableCrds[VirtualMachineCrdName],
						false,
					),
					util.EventTypePredicate(
						!r.AvailableCrds[StorageClientCrdName],
						false,
						r.AvailableCrds[StorageClientCrdName],
						false,
					),
				),
```

Reply:
Then the reconcile can get trigerred for wrong reason, like `. Or` isn't scoped for the object which trigerred the event.

Basically we want an And b/n event obj name & event type (create/update/delete/generic etc)

Reply:
```
				predicate.Or(
					predicate.And(
						util.NamePredicate(VirtualMachineCrdName),
						util.EventTypePredicate(
							!r.AvailableCrds[VirtualMachineCrdName],
							false,
							r.AvailableCrds[VirtualMachineCrdName],
							false,
						),
					),
					predicate.And(
						util.NamePredicate(StorageClientCrdName),
						util.EventTypePredicate(
							!r.AvailableCrds[StorageClientCrdName],
							false,
							r.AvailableCrds[StorageClientCrdName],
							false,
						),
					),
				),
```
I would prefer this over CrdCreateAndDeletePredicate, but I would let the other maintainers decide the right approach

Reply:
much better, meets the requirement, will change.

Reply:
done.

Reply:
ack, can resolve